Update the Enhancements role handbook to match the new opt-in process (#1517)
kubernetes/sig-release: release-team/role-handbooks/enhancements/README.md

### Access Required

> Ensure that the previous Enhancements Lead has given you access to:
> - Access to the previous Kubernetes Release Enhancements Tracking Sheet.

Reviewer: small nit:
```suggestion
Ensure that the previous Enhancements Lead has given you access to:
- The previous Kubernetes Release Enhancements Tracking Sheet.
```

- Verify issues have k/k PRs associated so they can be referenced and easily tracked. This is going to be critical come Enhancement Freeze and Code Freeze to see the status of the code.
- Work with the Release Lead to introduce yourself, talk about release information, and relay information about opting into the release with SIG Leads.
- Weekly Release meetings require updates of current status. Use the `Feature stats` tab to update the release team on counts of enhancements in good and bad progress.

#### Collection Monitoring and Triage (Week 2-3)

- Monitor the `KEP Collection` tab regularly for new additions as SIGs opt-in KEPs into the release.
- Move data from `KEP Collection` into the `data` tab to be tracked officially for the release.
- Manually add the KEP issue number into the `Enhancements` tab and verify that the KEP is now included.
  - Entering the KEP issue number will auto populate the Enhancement Name, SIG, KEP Assignee, Proposal, and KEP State.
  - Fill out rest of the column manually based on current status of the KEP.

> - For the opted in issues, apply the correct milestone, `tracked/yes` label, and `stage/xxx` label.

Reviewer 2: Should we enumerate the possible stage labels?
```suggestion
- For the opted in issues apply the correct milestone, `tracked/yes` label, and correct stage label (`stage/alpha`, `stage/beta`, or `stage/stable`) for the stage the enhancement is graduating to in this release.
```

Reviewer: As an extension to the above comment, do we need a `stage/deprecation` label as well? We might see more deprecations in the near future due to recent changes in policy like the prevent perma-beta policy.

Author: @JamesLaverack @palnabarun Can we leave it generic so that not all stages (currents and future) need to be added to the handbook?

As for the `stage/deprecation` label, that's a great idea but since that's nonexistent right now, I think we should keep that separate from this PR

Author: we can bring up the `stage/deprecation` in the next enhancements meeting and get that label added.

Reviewer: > Can we leave it generic so that not all stages (currents and future) need to be added to the handbook?

Sure. We can keep the statement as it was intended to be. 👍🏽

Reviewer: +1 from me on discussing this in the enhancements meeting.

Reviewer 2: If these are subject to change, as it sounds like they might soon, then +1 for leaving it generic.

- Start syncing with Communications Team on giving an induction what's coming up for the release.
- Send an email to [Kubernetes-Dev](https://groups.google.com/forum/#!forum/kubernetes-dev) that Enhancement freeze is coming and share current Enhancements status. Examples [1](https://groups.google.com/g/kubernetes-dev/c/-nTNtBBHL2Y/m/WfNzb_E1EAAJ).

#### Enhancements Freeze (Week 4)

- On Freeze day, send an email to [Kubernetes-Dev](https://groups.google.com/forum/#!forum/kubernetes-dev) that freeze has happened and upcoming key dates. Examples [1](https://groups.google.com/g/kubernetes-dev/c/JDM7bNKvhqQ/m/8S7BXtXPBQAJ).
- Remove [SIG Release Leads Google Group][sig-release-leads-group]'s access to the `KEP Collection` sheets on the enhancement collection sheet.
- Remove any enhancements that failed to meet the criteria by the Enhancement freeze deadline. 
    - Set their status in the sheet to `Removed from Milestone` and use the `Enhancements` -> `Remove Enhancements from Milestone` menu option to move them over to the `Removed from milestone` tab.

>     - Remove the milestone and change `tracked/yes` label to `tracked/no`

Reviewer: ```suggestion
    - Remove the milestone and change `tracked/yes` label to `tracked/no` on the enhancement issue.
```

    - Create an exception file in the Release for exceptions Example [1](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.14/exception.yaml).
    - If a previously removed Enhancement has had their exception Approved, set their status to `Tracked` and use the `Enhancements` -> `Track Removed Enhancements` menu option to move it back to the `Enhancements` and `Docs` tabs.

#### Post Enhancements Freeze (Week 5-10)

- Stay on top of issues and continually monitor them twice a week and look at attached PRs. As Code Freeze gets closer, if there are PRs that have not been merged, move the issue to `At Risk`. If there is no activity, ping issue owners on either the issue or the k/k PR.
- Monitor issues that are `At Risk` closely, almost daily. Code Freeze means no new code and keeping tabs on the status of the k/k PR is critical to planning. Make decisions if the enhancement should be deferred and work with SIG Leads to determine the best path forward.

#### Code Freeze (Week 10+)

- Remove any enhancements that failed to merge their code by the Code freeze deadline.
  - Set their status in the sheet to `Removed from Milestone`.

>   - Remove the milestone and change `tracked/yes` label to `tracked/no`.

Reviewer: ```suggestion 
  - Remove the milestone and change `tracked/yes` label to `tracked/no` on the enhancement issue.
```

  - Remove the milestone from all open `k/k` PRs related to the enhancement.
- Any enhancements removed from the milestone will now require an exception. As exception requests come in, discuss each with the Release Lead (and Shadows) to arrive at an approve/reject decision.

>   - Add a `/hold` label to all incoming exceptions to prevent from accidental merge.

Reviewer: The exceptions come as an email unless we are changing the exceptions process. Where do we put the `/hold`?

  - Add incoming exception information to the previous created `exception.yaml` file.
  - If a previously removed Enhancement has had their exception Approved, set their status back to `Tracked`

> - Start planning for the next release while assisting the Release Lead with anything relating to Analyst or Public Relation planning. Work with the Communications Lead to develop major themes for the official Kubernetes blog post.

Reviewer: ```suggestion
- Start planning for the next release while assisting the Release Lead with anything relating to analytics or Public Relation planning of the release. Work with the Communications Lead to develop major themes for the official Kubernetes blog post.
```

- Verify issues have k/k PRs associated so they can be referenced and easily tracked. This is going to be critical come Enhancement Freeze and Code Freeze to see the status of the code.
- Work with the Release Lead to introduce yourself, talk about release information, and relay information about opting into the release with SIG Leads.
- Weekly Release meetings require updates of current status. Use the `Feature stats` tab to update the release team on counts of enhancements in good and bad progress.

#### Collection Monitoring and Triage (Week 2-3)

> - Move data from `KEP Collection` into the `data` tab to be tracked officially for the release.

Reviewer 3: I think something like: "Monitor the `KEP Collection` tab regularly to new additions as SIGs enter opted-in KEPs into the release"  just a smidge more explicit about why they are checking that tab

Reviewer 3: Also something like: Move data from `KEP Collection` into the `data` tab to be tracked officially for the release and verify that the KEP is now included in the `Enhancements` tab.

Author: done!

#### Code Freeze (Week 10+)

- Remove any enhancements that failed to merge their code by the Code freeze deadline.
  - Set their status in the sheet to `Removed from Milestone`.
  - Remove the milestone and change `tracked/yes` label to `tracked/no`.
  - Remove the milestone from all open `k/k` PRs related to the enhancement.
- Any enhancements removed from the milestone will now require an exception. As exception requests come in, discuss each with the Release Lead (and Shadows) to arrive at an approve/reject decision.
  - Add a `/hold` label to all incoming exceptions to prevent from accidental merge.
  - Add incoming exception information to the previous created `exception.yaml` file.
  - If a previously removed Enhancement has had their exception Approved, set their status back to `Tracked`
- Start planning for the next release while assisting the Release Lead with anything relating to Analyst or Public Relation planning. Work with the Communications Lead to develop major themes for the official Kubernetes blog post.

### Working with the Enhancement Tracking Sheet

#### Cross Release Enhancement Tracking

> The source of truth for Enhancements is the `data` tab within the tracking sheet. All other tabs are driven off the data in this tab.

Reviewer 3: maybe note that certain updates should be made in this tab not the enhancements tab?

Author: Listed the columns that should be changed in the `data` tab

Author: also added to L124
